CardanoSharp.Wallet/CIPs/CIP30/Interfaces/IWebWalletApi.cs

```diff
@@ -0,0 +1,155 @@
+﻿using CardanoSharp.Wallet.CIPs.CIP30.Models;
+using System.Threading.Tasks;
+
+namespace CardanoSharp.Blazor.Components.Interfaces
+{
+	/// <summary>
+	/// CIP30 Compliant full interface as per https://cips.cardano.org/cips/cip30/
+	/// </summary>
+	public interface IWebWalletApi
+	{
+		/// <summary>
+		/// api.getNetworkId(): Promise\
+		/// Errors: APIError
+		/// </summary>
+		/// <returns>
+		/// Returns the network id of the currently connected account. 0 is testnet and 1 is mainnet but other networks
+		/// can possibly be returned by wallets. Those other network ID values are not governed by this document.
+		/// This result will stay the same unless the connected account has changed.
+		/// </returns>
+		Task<int> GetNetworkId();
+
+		/// <summary>
+		/// api.getUtxos(amount: cbor\ = undefined, paginate: Paginate = undefined): Promise\
+		/// Errors: APIError, PaginateError
+		/// </summary>
+		/// <param name="amount">A hex-encoded string representing CBOR</param>
+		/// <param name="paginate"></param>
+		/// <returns>
+		/// If amount is undefined, this shall return a list of all UTXOs(unspent transaction outputs) controlled
+		/// by the wallet.If amount is not undefined, this request shall be limited to just the UTXOs that are required
+		/// to reach the combined ADA/multiasset value target specified in amount, and if this cannot be attained,
+		/// null shall be returned.The results can be further paginated by paginate if it is not undefined.
+		/// </returns>
+		Task<string[]> GetUtxos(string? amount = null, Paginate paginate = null);
```

Comment:
Would there be any negative implications of using strongly typed return types instead of CBOR hexes (e.g. Utxo[] in this case)?

Reply:
i guess this is just keeping it closer to the spec and to what actually comes out of the JS object in the dom... its easy enough to use the extension method i added to transform the cbor into a utxo object. for the latest version of the wallet connector the component supports inputs and outputs that are either structured or cbor strings. i dont have a really strong opinion either way

Reply:
That's a fair point, keeping an interface like-for-like with the spec makes sense